symphony-bdk-http/symphony-bdk-http-webclient/src/main/java/com/symphony/bdk/http/webclient/ApiClientWebClient.java

```diff
@@ -0,0 +1,287 @@
+package com.symphony.bdk.http.webclient;
+
+import com.symphony.bdk.http.api.ApiClient;
+import com.symphony.bdk.http.api.ApiException;
+import com.symphony.bdk.http.api.ApiResponse;
+import com.symphony.bdk.http.api.Pair;
+import com.symphony.bdk.http.api.util.TypeReference;
+
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+import java.io.File;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ApiClientWebClient implements ApiClient {
+
+  protected WebClient webClient;
+  protected String basePath;
+  protected Map<String, String> defaultHeaderMap;
+
+  public ApiClientWebClient(final WebClient webClient, String basePath, Map<String, String> defaultHeaders) {
+    this.webClient = webClient;
+    this.basePath = basePath;
+    this.defaultHeaderMap = new HashMap<>(defaultHeaders);
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public <T> ApiResponse<T> invokeAPI(
```

Comment:
I would name it invokeApi

Reply:
Used by generator code, we cannot change the contract. Well, yes we could but we would need to modify the mustache templates